pkg/store/bucket.go

```diff
@@ -2235,45 +2264,63 @@ func newBucketChunkReader(ctx context.Context, block *bucketBlock) *bucketChunkR
 		ctx:      ctx,
 		block:    block,
 		stats:    &queryStats{},
-		preloads: make([][]uint32, len(block.chunkObjs)),
+		preloads: make([][]preloadIdx, len(block.chunkObjs)),
 		chunks:   map[uint64]chunkenc.Chunk{},
 	}
 }
 
-// addPreload adds the chunk with id to the data set that will be fetched on calling preload.
-func (r *bucketChunkReader) addPreload(id uint64) error {
+func (r *bucketChunkReader) Chunk(id uint64) (chunkenc.Chunk, error) {
```

Comment:
`Chunk` and `Close` methods are unchanged; just moved exported functions closer to receiver definition.

Reply:
Thanks!

Reply:
Actually, scratch that. First of all, I've broken `Chunk()` while changing preload flow. Secondly, `Chunk()` is no longer used, so I've deleted it altogether.